### PR TITLE
Message board final

### PR DESCRIPTION
**Pull request by smile0711**
Finally think it is done.  There is one new NPM html-sanitize.

### CODE REVIEW

**Comment by smile0711** [comment]
There might be a glitch with the meta "wrapper" tag.  While logging in, the plaza chat and 3d world does not load.  It requires a refresh to load.

File: spa/src/pages/MessageBoard.vue [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+        </div>
+        <div>
+          <h2>Post a Message</h2>
+        </div>
+        <div class="text-sm text-yellow-200 w-5/12 justify-center border-black border-4">
+          Some HTML coding has been blocked for security reasons.  Basic HTML tags
+          (i.e. &lt;p&gt;, &lt;br&gt;, &lt;a href&gt;, and &lt;img src&gt;) are
+          allowed.  If a disallowed tag is used, an error message will display.
+        </div>
+        <lable for="subject">Subject:</lable>&nbsp;&nbsp;
```
Comment (dburleson) [inline spa/src/pages/MessageBoard.vue:80]: `lable` should be `label`

Comment (dburleson) [reply spa/src/pages/MessageBoard.vue:80]: multiple occurrences on this script

File: api/db/migrations/20230409194209_add_messageboard.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      table.foreign('member_id')
+        .references('member.id');
+            
+      table.text('subject')
+        .notNullable();
+            
+      table.text('message')
+        .notNullable();
+            
+      table.integer('parent_id', 11)
```
Comment (dburleson) [inline api/db/migrations/20230409194209_add_messageboard.ts:34]: When trying to create a new message (not a reply), I receive an error that "parent_id" does not have a default value.

**Review by dburleson — CHANGES_REQUESTED** [review]
Unable to proceed with testing as I could not post a message. Left to test:

* posting a message
* posting a reply
* reading a message
* code tags

File: spa/src/App.vue
[the remark this answers is not in the document]
Comment (smile0711) [reply spa/src/App.vue:1]: This was my attempt to take away the panels.  Also this was attempt to not populate chatrooms for each open message board.  However, it is effecting the loading of the places, another way maybe needed.

File: api/db/migrations/20230409194209_add_messageboard.ts (hunk 2 above)
Comment (smile0711) [reply api/db/migrations/20230409194209_add_messageboard.ts:34]: I may have forgotten to update the migration to set a default null

File: spa/src/pages/MessageBoard.vue (hunk 1 above)
Comment (smile0711) [reply spa/src/pages/MessageBoard.vue:80]: Late night copying and pasting 😄

**Pull request closed without merging**